src/libs/ReportTitleUtils.ts

```diff
@@ -0,0 +1,112 @@
+    return reportNameValuePairs?.expensify_text_title;
+}
+
+/**
+ * Update title field in report's rNVP to match the policy's title field configuration
+ * This is the JavaScript equivalent of the backend updateTitleFieldToMatchPolicy function
+ */
+function updateTitleFieldToMatchPolicy(reportID: string, policy?: Policy): OnyxUpdate[] {
+    if (!reportID || !policy) {
+        return [];
+    }
+
+    // Get the policy's title field configuration
+    const policyTitleField = getTitleReportField(policy.fieldList ?? {});
+
+    // Early return if policy doesn't have a title field
+    if (!policyTitleField) {
+        return [];
+    }
+
+    // Create the update to set/update the title field in rNVP
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_NAME_VALUE_PAIRS}${reportID}`,
+            value: {
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                expensify_text_title: policyTitleField,
+            },
+        },
+    ];
+
+    return optimisticData;
+}
+
+function updateTitleFieldWithExactValue(reportID: string, policyTitleField: string): OnyxUpdate[] {
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_NAME_VALUE_PAIRS}${reportID}`,
+            value: {
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                expensify_text_title: policyTitleField,
+            },
+        },
+    ];
+
+    return optimisticData;
+}
+
+/**
+ * Remove title field from report's rNVP when report is manually renamed
+ * This indicates the report should preserve its custom name
+ */
+function removeTitleFieldFromReport(reportID: string): OnyxUpdate[] {
+    if (!reportID) {
+        return [];
+    }
+
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_NAME_VALUE_PAIRS}${reportID}`,
+            value: {
+                // eslint-disable-next-line @typescript-eslint/naming-convention
+                expensify_text_title: null,
+            },
+        },
+    ];
+
+    return optimisticData;
+}
+
+/**
+ * Check if a report should have its title field updated based on conditions
+ */
+function shouldUpdateTitleField(report: Report): boolean {
+    // todo: this should be more sophisticated function. check for iou etc
+    if (!report) {
+        return false;
+    }
+    // Skip chat reports
+    if (isChatReport(report)) {
+        return false;
+    }
+
+    // Skip if report has statement card ID (backend check: !getValue(db, reportID, NVP_STATEMENT_CARD_ID).empty())
+    // This would need to be implemented based on how statement card IDs are stored in the frontend
+
+    const policyTitleField = getTitleFieldFromRNVP(report.reportID);
```

Comment:
Can we please call this report title field since it comes from the report name value pairs and not the policy?